client.go

```diff
@@ -1343,119 +1345,13 @@ func (c *HostClient) doNonNilReqResp(req *Request, resp *Response) (bool, error)
 			req.Header.userAgent = append(req.Header.userAgent[:], userAgent...)
 		}
 	}
-	if c.Transport != nil {
-		err := c.Transport(req, resp)
-		return err == nil, err
-	}
-
-	var deadline time.Time
-	if req.timeout > 0 {
-		deadline = time.Now().Add(req.timeout)
-	}
-
-	cc, err := c.acquireConn(req.timeout, req.ConnectionClose())
-	if err != nil {
-		return false, err
-	}
-	conn := cc.c
-
-	resp.parseNetConn(conn)
-
-	writeDeadline := deadline
-	if c.WriteTimeout > 0 {
-		tmpWriteDeadline := time.Now().Add(c.WriteTimeout)
-		if writeDeadline.IsZero() || tmpWriteDeadline.Before(writeDeadline) {
-			writeDeadline = tmpWriteDeadline
-		}
-	}
 
-	if err = conn.SetWriteDeadline(writeDeadline); err != nil {
-		c.closeConn(cc)
-		return true, err
+	// exec
```

Comment:
```suggestion
```

Reply:
I add a transport() method for HostClient to simplify the code.